lib/pbench/server/api/resources/query_apis/query_dataset_list.py

```diff
@@ -0,0 +1,290 @@
+from flask import request, jsonify
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser
+from pbench.server.api.resources.query_apis import (
+    get_es_url,
+    get_index_prefix,
+    gen_month_range,
+    get_user_term,
+)
+
+
+class QueryDatasetList(Resource):
+    """
+    Abstracted Pbench API to get date-bounded dataset data for a particular
+    controller.
+
+    Note that although the name "queryResults" is retained for consistency
+    with the existing dashboard UI code, the word "results" is a bit
+    misleading in that this deals with Pbench "run" documents, not Pbench
+    "results" documents. It's a follow-on to queryControllers, which returns
+    the list of controllers having datasets within a specified date range. The
+    queryResults API returns the list of dataset names corresponding to
+    specified controller within the date range.
+    """
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def post(self):
+        """
+        POST for Pbench datasets from a specified controller within the
+        specified date range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
```

Comment:
Why are we requiring that the API invocation provide a "user" argument?

Why not make this parameter optional to add a constraint on which data is visible.  But we should not use that parameter for access control.

Instead, we have to get the effective user of the query, and make sure the query results in data that is "public" or from a user that matches this user.

Reply:
And "email-tag" should become "username" or similar.